Epic/projects managment

waleedahmad2 opened the pull request:
# Description

- Implemented modular sidebar architecture - Created SidebarHeader, SidebarItemsList, SidebarListItem, and SidebarFooter components with proper separation of concerns
- Set up nested routing with Outlet - Configured React Router to use Layout component with Outlet for nested routes (Projects, Boards, Tasks)
- Added persistent navigation state - Implemented localStorage-based selection persistence so sidebar state survives hard reloads
- Converted components to arrow functions - Refactored all sidebar components to use arrow function syntax with implicit returns where appropriate
- Established project structure - Created placeholder pages for Projects, Boards, and Tasks with proper routing configuration and default redirect to Projects page

## Type of change
- [x] New feature (non-breaking change which adds functionality)



## Trello Ticket link (if applicable)
https://app.clickup.com/t/86eutrgg8


## How Has This Been Tested?

<!-- Describe the tests you ran and how -->

- [x] Unit Tests
- [ ] Manual Testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [x] My branch follows the naming convention: `type/description` (e.g., feature/add-login, fix/memory-leak)
- [x] I have verified the UI (if applicable)

waleedahmad2 committed:
chore: update lucide-react dependency and refactor sidebar components for improved navigation structure

waleedahmad2 committed:
refactor: standardize component formatting and improve code consistency across sidebar and layout components

waleedahmad2 committed:
refactor: adjust import order for improved readability in Layout component

waleedahmad2 committed:
refactor: update component exports and rename page components for clarity

waleedahmad2 committed:
refactor: streamline import statements and clean up component formatting in Layout

waleedahmad2 committed:
refactor: replace native button with custom Button component in SidebarFooter for consistency

waleedahmad2 committed:
refactor: remove duplicate Button import in SidebarFooter for cleaner code

Hammadhammad2 commented on src/components/layout/Layout.tsx:
```diff
@@ -0,0 +1,27 @@
+ * Provides sidebar navigation and main content area using Outlet
+ */
+const Layout = (): JSX.Element => {
+  return (
```
Repeated mistakes

Hammadhammad2 commented on src/components/sidebar/SidebarItemsList.tsx:
```diff
@@ -0,0 +1,58 @@
+    navigate(path);
+  };
+
+  const persistedSelectedPath = useMemo<string | null>(() => {
```
unnecassary

Hammadhammad2 commented on src/components/sidebar/SidebarItemsList.tsx:
```diff
@@ -0,0 +1,58 @@
+  const { pathname } = useLocation();
+
+  const handleNavigation = (path: string): void => {
+    localStorage.setItem(SELECTED_KEY, path);
```
use the functions from utils

waleedahmad2 committed:
Feat/project listings (#2)

* feat: add tooltip component and enhance UI with new AppHeader, SearchInput, and NoData components

* feature : restructure GenericTable component and implement new Pagination structure with sub-components for improved modularity,

* refactor: simplify ProjectStatusColumns component by consolidating props destructuring for improved readability

* refactor: update README to reflect new task management application features and improve project structure documentation

* refactor: clean up formatting in Projects component for improved readability

* feat: add new components and update imports for improved modularity and functionality in the project

* refactor: streamline destructuring of useProjects hook in Projects component for improved readability

* refactor: remove Pagination component and its related files to streamline common components and improve modularity

* refactor: enhance project components by improving destructuring and readability, and introduce STATUS_DOT_COLORS for better status representation

* refactor: improve formatting and destructuring in ProjectStatusColumns component for enhanced readability

* feat : ceated range() utility  function in src/utils/utils.ts to generate number arrays (0 to length-1).
Updated skeleton components (KanbanColumnsSkeleton, TableSkeleton) to use range(n).map() instead of verbose Array.from({ length: n }).map()

* refactor: streamline GenericTable and KanbanGroupedBoard components by improving destructuring and formatting for enhanced readability

* refactor: remove unnecessary whitespace in KanbanGroupedBoard component for improved code cleanliness

* fix : enhance AppHeader and GenericTable components by improving conditional rendering and destructuring for better readability

* refactor: remove unnecessary whitespace in GenericTable component for improved code cleanliness and readability

* refactor: enhance ProjectsTable and ProjectStatusColumns components by introducing PROJECT_STATUSES and PROJECT_TABLE_COLUMNS for improved readability and maintainability

* chore: update Node.js version in README and add skeleton loading styles in CSS; refactor components to utilize VIEW_TYPES and improve code readability

* refactor: update project management components to utilize new hooks and improve API integration; remove unused code and enhance readability
